DIEGO FELIPE CARVAJAL LOMBO

UNIVERSIDAD DE LOS ANDES

# **1. Importación de las librerías**

In [1100]:
# Para visualizar un reporte de los datos
from pandas_profiling import ProfileReport
# Preprocesar
from sklearn.preprocessing import StandardScaler, Normalizer
# Declaración de la semilla para el manejo de aleatoreidad
seed = 161
import pandas as pd
pd.set_option('display.max_columns', 90) # Número máximo de columnas a mostrar
pd.set_option('display.max_rows', 90) # Numero máximo de filas a mostar
# Libreria Numpy
import numpy as np
# Librerías para la visualización
import matplotlib.pyplot as plt
from matplotlib import cm
from mpl_toolkits.mplot3d import Axes3D
import scipy.stats as stats
# Seaborn
import seaborn as sns; sns.set()
np.random.seed(seed)

# **2. Exploración de los datos**

En está etapa se realiza el proceso correspondiente a la exploración y perfilimiento de datos dados de la FIFA con el fin de comprender que datos tenemos y sus propiedades

## **2.1 Carga de datos de la FIFA**

In [1101]:
# Ubicación de la base de datos de la FIFA
db_location = "/content/data_FIFA.csv"
# Leer los datos de la FIFA
df = pd.read_csv(db_location, sep=",", low_memory=False)

In [1102]:
# Dimensiones de los datos
df.shape

(18207, 86)

In [1103]:
#Ver los datos
df.sample(5)

Unnamed: 0      ID         Name  Age Nationality  Overall  Potential  \
3495         3495  209756   L. Jiménez   29    Colombia       72         72   
7836         7836  227370  M. Zetterer   22     Germany       67         75   
6323         6323  190648    S. Sirigu   29     Germany       69         69   
7353         7353  238008   J. Elliott   22     England       68         73   
11829       11829  219337     J. Calle   27    Colombia       64         64   

                         Club  Value  Wage  Special Preferred Foot  \
3495               Club Atlas  €2.4M  €15K     1737          Right   
7836         SV Werder Bremen  €850K   €5K     1090          Right   
6323          SV Darmstadt 98  €800K   €9K     1815          Right   
7353       Philadelphia Union    €1M   €3K     1400          Right   
11829  Independiente Medellín  €475K   €1K     1752           Left   

       International Reputation  Weak Foot  Skill Moves       Work Rate  \
3495                        1.0        3.0          2.0       Low/ High   
7836                        1.0        4.0          1.0  Medium/ Medium   
6323                        1.0        2.0          3.0    High/ Medium   
7353                        1.0        3.0          2.0       Low/ High   
11829                       1.0        3.0          3.0  Medium/ Medium   

      Body Type Real Face Position  Jersey Number        Joined Loaned From  \
3495     Normal        No      LCB            3.0   Jun 8, 2016         NaN   
7836     Stocky        No       GK           16.0  Jan 27, 2015         NaN   
6323     Normal        No       RB           17.0   Jul 1, 2013         NaN   
7353       Lean        No       CB            3.0  Feb 24, 2017         NaN   
11829    Normal        No      CAM           20.0  Apr 19, 2018         NaN   

      Contract Valid Until Height  Weight    LS    ST    RS    LW    LF    CF  \
3495                  2022    6'1  170lbs  58+2  58+2  58+2  54+2  55+2  55+2   
7836                  2019    6'2  174lbs   NaN   NaN   NaN   NaN   NaN   NaN   
6323                  2019    6'0  174lbs  65+2  65+2  65+2  67+2  66+2  66+2   
7353                  2018    6'5  181lbs  43+2  43+2  43+2  42+2  43+2  43+2   
11829                 2018    5'9  146lbs  61+2  61+2  61+2  63+2  62+2  62+2   

         RF    RW   LAM   CAM   RAM    LM   LCM    CM   RCM    RM   LWB   LDM  \
3495   55+2  54+2  54+2  54+2  54+2  55+2  57+2  57+2  57+2  55+2  64+2  66+2   
7836    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
6323   66+2  67+2  65+2  65+2  65+2  69+2  63+2  63+2  63+2  69+2  68+2  63+2   
7353   43+2  42+2  45+2  45+2  45+2  47+2  53+2  53+2  53+2  47+2  58+2  64+2   
11829  62+2  63+2  63+2  63+2  63+2  63+2  61+2  61+2  61+2  63+2  59+2  57+2   

        CDM   RDM   RWB    LB   LCB    CB   RCB    RB  Crossing  Finishing  \
3495   66+2  66+2  64+2  66+2  71+2  71+2  71+2  66+2      40.0       47.0   
7836    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN      17.0       16.0   
6323   63+2  63+2  68+2  67+2  63+2  63+2  63+2  67+2      68.0       57.0   
7353   64+2  64+2  58+2  60+2  66+2  66+2  66+2  60+2      21.0       21.0   
11829  57+2  57+2  59+2  58+2  56+2  56+2  56+2  58+2      64.0       55.0   

       HeadingAccuracy  ShortPassing  Volleys  Dribbling  Curve  FKAccuracy  \
3495              72.0          60.0     40.0       55.0   40.0        42.0   
7836              18.0          24.0     16.0       17.0   14.0        14.0   
6323              59.0          66.0     49.0       71.0   53.0        41.0   
7353              65.0          63.0     24.0       44.0   25.0        21.0   
11829             61.0          64.0     49.0       63.0   48.0        50.0   

       LongPassing  BallControl  Acceleration  SprintSpeed  Agility  \
3495          53.0         55.0          68.0         71.0     64.0   
7836          18.0         23.0          39.0         44.0     32.0   
6323          57.0         66.0          87.0         

In [1104]:
df.head()

Unnamed: 0      ID               Name  Age Nationality  Overall  Potential  \
0           0  158023           L. Messi   31   Argentina       94         94   
1           1   20801  Cristiano Ronaldo   33    Portugal       94         94   
2           2  190871          Neymar Jr   26      Brazil       92         93   
3           3  193080             De Gea   27       Spain       91         93   
4           4  192985       K. De Bruyne   27     Belgium       91         92   

                  Club    Value   Wage  Special Preferred Foot  \
0         FC Barcelona  €110.5M  €565K     2202           Left   
1             Juventus     €77M  €405K     2228          Right   
2  Paris Saint-Germain  €118.5M  €290K     2143          Right   
3    Manchester United     €72M  €260K     1471          Right   
4      Manchester City    €102M  €355K     2281          Right   

   International Reputation  Weak Foot  Skill Moves       Work Rate  \
0                       5.0        4.0          4.0  Medium/ Medium   
1                       5.0        4.0          5.0       High/ Low   
2                       5.0        5.0          5.0    High/ Medium   
3                       4.0        3.0          1.0  Medium/ Medium   
4                       4.0        5.0          4.0      High/ High   

    Body Type Real Face Position  Jersey Number        Joined Loaned From  \
0       Messi       Yes       RF           10.0   Jul 1, 2004         NaN   
1  C. Ronaldo       Yes       ST            7.0  Jul 10, 2018         NaN   
2      Neymar       Yes       LW           10.0   Aug 3, 2017         NaN   
3        Lean       Yes       GK            1.0   Jul 1, 2011         NaN   
4      Normal       Yes      RCM            7.0  Aug 30, 2015         NaN   

  Contract Valid Until Height  Weight    LS    ST    RS    LW    LF    CF  \
0                 2021    5'7  159lbs  88+2  88+2  88+2  92+2  93+2  93+2   
1                 2022    6'2  183lbs  91+3  91+3  91+3  89+3  90+3  90+3   
2                 2022    5'9  150lbs  84+3  84+3  84+3  89+3  89+3  89+3   
3                 2020    6'4  168lbs   NaN   NaN   NaN   NaN   NaN   NaN   
4                 2023   5'11  154lbs  82+3  82+3  82+3  87+3  87+3  87+3   

     RF    RW   LAM   CAM   RAM    LM   LCM    CM   RCM    RM   LWB   LDM  \
0  93+2  92+2  93+2  93+2  93+2  91+2  84+2  84+2  84+2  91+2  64+2  61+2   
1  90+3  89+3  88+3  88+3  88+3  88+3  81+3  81+3  81+3  88+3  65+3  61+3   
2  89+3  89+3  89+3  89+3  89+3  88+3  81+3  81+3  81+3  88+3  65+3  60+3   
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
4  87+3  87+3  88+3  88+3  88+3  88+3  87+3  87+3  87+3  88+3  77+3  77+3   

    CDM   RDM   RWB    LB   LCB    CB   RCB    RB  Crossing  Finishing  \
0  61+2  61+2  64+2  59+2  47+2  47+2  47+2  59+2      84.0       95.0   
1  61+3  61+3  65+3  61+3  53+3  53+3  53+3  61+3      84.0       94.0   
2  60+3  60+3  65+3  60+3  47+3  47+3  47+3  60+3      79.0       87.0   
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN      17.0       13.0   
4  77+3  77+3  77+3  73+3  66+3  66+3  66+3  73+3      93.0       82.0   

   HeadingAccuracy  ShortPassing  Volleys  Dribbling  Curve  FKAccuracy  \
0             70.0          90.0     86.0       97.0   93.0        94.0   
1             89.0          81.0     87.0       88.0   81.0        76.0   
2             62.0          84.0     84.0       96.0   88.0        87.0   
3             21.0          50.0     13.0       18.0   21.0        19.0   
4             55.0          92.0     82.0       86.0   85.0        83.0   

   LongPassing  BallControl  Acceleration  SprintSpeed  Agility  Reactions  \
0         87.0         96.0          91.0         86.0     91.0       95.0   
1         77.0         94.0          89.0         91.0     87.0       96.0   
2         78.0         95.0          94.0         90.0     96.0       94.0   
3         51.0         42.0          57.0         58.0     60.0       90.0   
4         91.0   

## **2.2 Descripción de los datos**

In [1105]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18207 entries, 0 to 18206
Data columns (total 86 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                18207 non-null  int64  
 1   ID                        18207 non-null  int64  
 2   Name                      18207 non-null  object 
 3   Age                       18207 non-null  int64  
 4   Nationality               18207 non-null  object 
 5   Overall                   18207 non-null  int64  
 6   Potential                 18207 non-null  int64  
 7   Club                      17966 non-null  object 
 8   Value                     18207 non-null  object 
 9   Wage                      18207 non-null  object 
 10  Special                   18207 non-null  int64  
 11  Preferred Foot            18159 non-null  object 
 12  International Reputation  18159 non-null  float64
 13  Weak Foot                 18159 non-null  float64
 14  Skill 

# **3. Limpieza de los datos**

In [1106]:
df.shape

(18207, 86)

Dada la exploración a los datos hecha previamente, se pudo ver que hay tres columnas que no respresentan nada para el negocio ni para el modelo.

La primera es 'ID', a pesar de que es importante para poder busca a un futbolista de manera eficiente, al momento de interpretar los datos esta no nos aporta nada, por lo que se va a eliminar.

La segunda es 'Unnamed: 0', esta se ve que es el número del registro en la tabla de datos. Nuevamente, esto para el negocio y para el análisis no nos aporta nada, por lo que se va a eliminar.

La tercera es 'Name', esta, a pesar de que es otra buena forma de buscar, no aporta nada al modelo ni al negocio, por lo que también se elimina.

In [1107]:
df = df.drop(['Unnamed: 0', 'ID', 'Name'], axis = 1)
df.shape

(18207, 83)

Vemos que el número de columnas disminuyó en 3.

Ahora, dado que no se tienen columnas que representen un posible ID, por ende, hagan que cada registro sea único, se va a proceder a eliminar los duplicados de la base de datos. Esto es debido a que los duplicados no aportan información nueva al negocio.

In [1108]:
df = df.drop_duplicates()
df.shape

(18207, 83)

Ningún dato era duplicado



Procedemos a mirar cuántos datos nulos hay en cada columna/atributo

In [1109]:
df.isnull().sum()

Age                             0
Nationality                     0
Overall                         0
Potential                       0
Club                          241
Value                           0
Wage                            0
Special                         0
Preferred Foot                 48
International Reputation       48
Weak Foot                      48
Skill Moves                    48
Work Rate                      48
Body Type                      48
Real Face                      48
Position                       60
Jersey Number                  60
Joined                       1553
Loaned From                 16943
Contract Valid Until          289
Height                         48
Weight                         48
LS                           2085
ST                           2085
RS                           2085
LW                           2085
LF                           2085
CF                           2085
RF                           2085
RW            

De la anterior tabla podemos ver que hay una columna 'Loaned From' cuya cantidad de elementos nulos es muy grande (casi a un 80%), por lo que esta no representa casi nada al modelo, pero si se eliminan estos registros, se estaría eliminando casi toda la base de datos, por lo que se decide eliminar únicamente la columna.

In [1110]:
df = df.drop(['Loaned From'], axis = 1)
df.shape

(18207, 82)

Gracias a la cantidad de elementos nulos (o vacíos) podemos ver que hay un grupo importante a tener en cuenta.

Este grupo corresponde a la puntuación en la posición del jugador, vemos que todas las columnas que contienen información de las posiciones del jugador tienen un total de 2085 registros nulos, por lo que es muy probable que estos registros sean los mismos.

En dado caso que los valores nulos de los atributos de la puntuación en la posición no coincidan, si se eliminan los registros, es muy posible que se esten eliminando una gran parte de la base de datos, por lo que primero se debe verificar que estos conjuntos no sean disyuntos.

In [1111]:
#Obtenemos todos los registros cuyo valor en 'LS' es nulo
df_nulos_LS = df[df['LS'].isnull()]
df_nulos_LS.head()

Age Nationality  Overall  Potential               Club   Value   Wage  \
3    27       Spain       91         93  Manchester United    €72M  €260K   
9    25    Slovenia       90         93    Atlético Madrid    €68M   €94K   
18   26     Germany       89         92       FC Barcelona    €58M  €240K   
19   26     Belgium       89         90        Real Madrid  €53.5M  €240K   
22   32     Germany       89         89  FC Bayern München    €38M  €130K   

    Special Preferred Foot  International Reputation  Weak Foot  Skill Moves  \
3      1471          Right                       4.0        3.0          1.0   
9      1331          Right                       3.0        3.0          1.0   
18     1328          Right                       3.0        4.0          1.0   
19     1311           Left                       4.0        2.0          1.0   
22     1473          Right                       5.0        4.0          1.0   

         Work Rate Body Type Real Face Position  Jersey Number        Joined  \
3   Medium/ Medium      Lean       Yes       GK            1.0   Jul 1, 2011   
9   Medium/ Medium    Normal       Yes       GK            1.0  Jul 16, 2014   
18  Medium/ Medium    Normal       Yes       GK           22.0   Jul 1, 2014   
19  Medium/ Medium  Courtois       Yes       GK            1.0   Aug 9, 2018   
22  Medium/ Medium    Normal       Yes       GK            1.0   Jul 1, 2011   

   Contract Valid Until Height  Weight   LS   ST   RS   LW   LF   CF   RF  \
3                  2020    6'4  168lbs  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
9                  2021    6'2  192lbs  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
18                 2022    6'2  187lbs  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
19                 2024    6'6  212lbs  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
22                 2021    6'4  203lbs  NaN  NaN  NaN  NaN  NaN  NaN  NaN   

     RW  LAM  CAM  RAM   LM  LCM   CM  RCM   RM  LWB  LDM  CDM  RDM  RWB   LB  \
3   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
9   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
18  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
19  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
22  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   

    LCB   CB  RCB   RB  Crossing  Finishing  HeadingAccuracy  ShortPassing  \
3   NaN  NaN  NaN  NaN      17.0       13.0             21.0          50.0   
9   NaN  NaN  NaN  NaN      13.0       11.0             15.0          29.0   
18  NaN  NaN  NaN  NaN      15.0       14.0             11.0          36.0   
19  NaN  NaN  NaN  NaN      14.0       14.0             13.0          33.0   
22  NaN  NaN  NaN  NaN      15.0       13.0             25.0          55.0   

    Volleys  Dribbling  Curve  FKAccuracy  LongPassing  BallControl  \
3      13.0       18.0   21.0        19.0         51.0         42.0   
9      13.0       12.0   13.0        14.0         26.0         16.0   
18     14.0       17.0   18.0        12.0         42.0         18.0   
19     12.0       13.0   19.0        20.0         35.0         23.0   
22     11.0       30.0   14.0        11.0         59.0         48.0   

    Acceleration  SprintSpeed  Agility  Reactions  Balance  ShotPower  \
3           57.0         58.0     60.0       90.0     43.0       31.0   
9           43.0         60.0     67.0       86.0     49.0       22.0   
18          38.0         50.0     37.0       85.0     43.0       22.0   
19          46.0         52.0     61.0       84.0     45.0       36.0   
22          54.0         60.0     51.0       84.0     35.0       25.0   

    Jumping  Stamina  Strength  LongShots  Aggression  Interceptions  \
3      67.0     43.0      64.0       12.0        38.0           30.0   
9      76.0     41.0      78.0       12.0        34.0           19.0   
18     79.0     35.0      79.0       10.0        43.0           22.0   
19    

Como vemos en la tabla anterior, la predicción de que los atributos que indican la posición de un jugador cuyo valor sea nulo son los mismos parece ser acertada.

In [1112]:
#Obtenemos todos los registros cuyo valor en 'ST' es nulo
df_nulos_ST = df[df['ST'].isnull()]
df_nulos_ST.head()

Age Nationality  Overall  Potential               Club   Value   Wage  \
3    27       Spain       91         93  Manchester United    €72M  €260K   
9    25    Slovenia       90         93    Atlético Madrid    €68M   €94K   
18   26     Germany       89         92       FC Barcelona    €58M  €240K   
19   26     Belgium       89         90        Real Madrid  €53.5M  €240K   
22   32     Germany       89         89  FC Bayern München    €38M  €130K   

    Special Preferred Foot  International Reputation  Weak Foot  Skill Moves  \
3      1471          Right                       4.0        3.0          1.0   
9      1331          Right                       3.0        3.0          1.0   
18     1328          Right                       3.0        4.0          1.0   
19     1311           Left                       4.0        2.0          1.0   
22     1473          Right                       5.0        4.0          1.0   

         Work Rate Body Type Real Face Position  Jersey Number        Joined  \
3   Medium/ Medium      Lean       Yes       GK            1.0   Jul 1, 2011   
9   Medium/ Medium    Normal       Yes       GK            1.0  Jul 16, 2014   
18  Medium/ Medium    Normal       Yes       GK           22.0   Jul 1, 2014   
19  Medium/ Medium  Courtois       Yes       GK            1.0   Aug 9, 2018   
22  Medium/ Medium    Normal       Yes       GK            1.0   Jul 1, 2011   

   Contract Valid Until Height  Weight   LS   ST   RS   LW   LF   CF   RF  \
3                  2020    6'4  168lbs  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
9                  2021    6'2  192lbs  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
18                 2022    6'2  187lbs  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
19                 2024    6'6  212lbs  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
22                 2021    6'4  203lbs  NaN  NaN  NaN  NaN  NaN  NaN  NaN   

     RW  LAM  CAM  RAM   LM  LCM   CM  RCM   RM  LWB  LDM  CDM  RDM  RWB   LB  \
3   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
9   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
18  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
19  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
22  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   

    LCB   CB  RCB   RB  Crossing  Finishing  HeadingAccuracy  ShortPassing  \
3   NaN  NaN  NaN  NaN      17.0       13.0             21.0          50.0   
9   NaN  NaN  NaN  NaN      13.0       11.0             15.0          29.0   
18  NaN  NaN  NaN  NaN      15.0       14.0             11.0          36.0   
19  NaN  NaN  NaN  NaN      14.0       14.0             13.0          33.0   
22  NaN  NaN  NaN  NaN      15.0       13.0             25.0          55.0   

    Volleys  Dribbling  Curve  FKAccuracy  LongPassing  BallControl  \
3      13.0       18.0   21.0        19.0         51.0         42.0   
9      13.0       12.0   13.0        14.0         26.0         16.0   
18     14.0       17.0   18.0        12.0         42.0         18.0   
19     12.0       13.0   19.0        20.0         35.0         23.0   
22     11.0       30.0   14.0        11.0         59.0         48.0   

    Acceleration  SprintSpeed  Agility  Reactions  Balance  ShotPower  \
3           57.0         58.0     60.0       90.0     43.0       31.0   
9           43.0         60.0     67.0       86.0     49.0       22.0   
18          38.0         50.0     37.0       85.0     43.0       22.0   
19          46.0         52.0     61.0       84.0     45.0       36.0   
22          54.0         60.0     51.0       84.0     35.0       25.0   

    Jumping  Stamina  Strength  LongShots  Aggression  Interceptions  \
3      67.0     43.0      64.0       12.0        38.0           30.0   
9      76.0     41.0      78.0       12.0        34.0           19.0   
18     79.0     35.0      79.0       10.0        43.0           22.0   
19    

Al obtener los registros cuyo valor es nulo en 'ST' vemos que obtenemos los mismos primeros 5 registros que al obtener los de 'LS'. Lo que hace intuir aún más que estos dos conjuntos de registros nulos son iguales. 

Además, podemos intuir que estos registros cuya posición es nula correspondan a los arqueros, esto tendría sentido dado que no se le coloque una valoración de las demás posiciones.

Para comprobar esto, vamos a hacer una copia de los datos en primer lugar.

In [1113]:
#Sacamos una copia de los datos
df_c = df
df_c.shape

(18207, 82)

Vemos que tiene las mismas cantidad de registros y atributos que el DF original.

Ahora, vamos a eliminar los resgitros cuyo valor en el atributo 'ST' sea nulo.

In [1114]:
df_c = df_c[df_c['LS'].notnull()]
df_c.shape

(16122, 82)

Vemos que ya obtuvimos todos los registros cuyo valor en el atributo 'ST' no es nulo en la copia. Lo podemos comprobar dado que la cantidad de registros restantes en la sobra (16122) más la cantidad de nulos de 'ST' (2085) da igual a la cantidad de registros en el DF original (18207).

Ahora, veremos la cantidad de nulos en la copia para los atributos que corresponden a la puntuacion en la posición del jugador.

In [1115]:
df_c.isnull().sum()

Age                            0
Nationality                    0
Overall                        0
Potential                      0
Club                         196
Value                          0
Wage                           0
Special                        0
Preferred Foot                 0
International Reputation       0
Weak Foot                      0
Skill Moves                    0
Work Rate                      0
Body Type                      0
Real Face                      0
Position                       0
Jersey Number                  0
Joined                      1371
Contract Valid Until         196
Height                         0
Weight                         0
LS                             0
ST                             0
RS                             0
LW                             0
LF                             0
CF                             0
RF                             0
RW                             0
LAM                            0
CAM       

De la anterior tabla podemos concluir que, los atributos de la puntuación en la posición, es decir, dede 'LS' hasta 'RB' cuyo valor sea nulo (2085 reggistros), corresponden a los mismos registros siempre, dado que al eliminar los registros con valor 'NaN' en el atributo 'LS', se eliminan también los nulos del conjunto de atrivutos antes mencionado, por lo que no representa un gran porcentaje al total de registros del DF original.

Con esta conclusión se procede a eliminar estos registros en el DF original.




In [1116]:
df = df[df['LS'].notnull()]
df.shape

(16122, 82)

Una solución que se consideró hacer a este conjunto de atributos cuyo valor era 'NaN' era intentar predecir este valor usando un modelo de regresión lineal. Pero esto fue descartado al momento de comprobar que los registros pertenecían al mismo conjunto de atributos nulos. Dado que es bastante más complejo predecir un valor de un registro cuando tiene muchos atributos nulos.

En segundo lugar se consideró poner la mediana o moda de DF total, pero dado que eran bastantes registros y atributos, se prefiere no inventar tantos datos.

La última solución que se pensó para este problema fue colocar un valor de 0 en todos estos atributos, ya que lo más posible es que estos registros correspondan a los arqueros, pero esto podría generar una cantidad muy grande de Outliers.

Luego de eliminar estos registros, vemos que todavía quedan ciertos registros cuyo valor es nulo, vemos que estos en total no superan los 1600 registros, es decir, son menores al 10% total del DF. Por lo que también se pueden eliminar.

In [1117]:
df = df.dropna()
df.isnull().sum()

Age                         0
Nationality                 0
Overall                     0
Potential                   0
Club                        0
Value                       0
Wage                        0
Special                     0
Preferred Foot              0
International Reputation    0
Weak Foot                   0
Skill Moves                 0
Work Rate                   0
Body Type                   0
Real Face                   0
Position                    0
Jersey Number               0
Joined                      0
Contract Valid Until        0
Height                      0
Weight                      0
LS                          0
ST                          0
RS                          0
LW                          0
LF                          0
CF                          0
RF                          0
RW                          0
LAM                         0
CAM                         0
RAM                         0
LM                          0
LCM       

In [1118]:
df.shape

(14743, 82)

Si hacemos la resta de la cantidad de registros antes de eliminar los nulos en el útlimo código con los resultantes a estos, vemos que este número (1379) coincide con la cantidad de jugadores que tienen vacío el atributo de 'Release Clause', es decir solo se eliminaron esa cantidad de registros.

Con esto, se han eliminado todos los registros que tengan un atributo nulo o vacío.

# **4. Exploración las columnas**

Se hace una transformación a los atributos que representan un valor numérico pero se guardan de manera diferente

Ahora que no tenemos registros nulos, se va a realizar una exploración de todas las columnas para así obtener mejores resultados en el dashboard.

In [1119]:
df.sample(5)

Age  Nationality  Overall  Potential            Club   Value  Wage  \
11537   35    Argentina       64         64    Curicó Unido   €190K   €1K   
13132   20       France       62         76   Chievo Verona   €675K   €2K   
270     25  Ivory Coast       82         85  Crystal Palace  €28.5M  €78K   
13329   23      Denmark       62         69       Aarhus GF   €350K   €2K   
1885    21      Morocco       75         83       AS Monaco    €10M  €39K   

       Special Preferred Foot  International Reputation  Weak Foot  \
11537     1665           Left                       1.0        3.0   
13132     1398          Right                       1.0        3.0   
270       1930          Right                       2.0        3.0   
13329     1715           Left                       1.0        2.0   
1885      1933          Right                       1.0        3.0   

       Skill Moves       Work Rate Body Type Real Face Position  \
11537          3.0  Medium/ Medium      Lean        No      LDM   
13132          2.0  Medium/ Medium    Normal        No       ST   
270            5.0       High/ Low      Lean       Yes       LS   
13329          2.0    High/ Medium    Normal        No       LB   
1885           3.0  Medium/ Medium    Normal        No       CM   

       Jersey Number       Joined Contract Valid Until Height  Weight    LS  \
11537           17.0  Jul 1, 2014                 2023    5'5  139lbs  61+2   
13132           11.0  Jul 1, 2017                 2020    6'1  172lbs  60+2   
270             11.0  Feb 2, 2015                 2023   5'11  146lbs  76+2   
13329           16.0  Feb 9, 2018                 2021    6'0  170lbs  57+2   
1885            15.0  Jul 1, 2016                 2021    6'0  165lbs  68+2   

         ST    RS    LW    LF    CF    RF    RW   LAM   CAM   RAM    LM   LCM  \
11537  61+2  61+2  65+2  65+2  65+2  65+2  65+2  66+2  66+2  66+2  64+2  63+2   
13132  60+2  60+2  54+2  57+2  57+2  57+2  54+2  53+2  53+2  53+2  52+2  47+2   
270    76+2  76+2  82+2  80+2  80+2  80+2  82+2  80+2  80+2  80+2  81+2  71+2   
13329  57+2  57+2  59+2  58+2  58+2  58+2  59+2  57+2  57+2  57+2  60+2  56+2   
1885   68+2  68+2  71+2  71+2  71+2  71+2  71+2  73+2  73+2  73+2  72+2  73+2   

         CM   RCM    RM   LWB   LDM   CDM   RDM   RWB    LB   LCB    CB   RCB  \
11537  63+2  63+2  64+2  56+2  57+2  57+2  57+2  56+2  53+2  51+2  51+2  51+2   
13132  47+2  47+2  52+2  39+2  37+2  37+2  37+2  39+2  38+2  38+2  38+2  38+2   
270    71+2  71+2  81+2  60+2  55+2  55+2  55+2  60+2  55+2  45+2  45+2  45+2   
13329  56+2  56+2  60+2  61+2  57+2  57+2  57+2  61+2  60+2  58+2  58+2  58+2   
1885   73+2  73+2  72+2  69+2  70+2  70+2  70+2  69+2  68+2  66+2  66+2  66+2   

         RB  Crossing  Finishing  HeadingAccuracy  ShortPassing  Volleys  \
11537  53+2      72.0       58.0             45.0          73.0     71.0   
13132  38+2      36.0       68.0             68.0          42.0     57.0   
270    55+2      81.0       77.0             47.0          75.0     61.0   
13329  60+2      64.0       48.0             41.0          58.0     42.0   
1885   68+2      64.0       56.0             58.0          78.0     62.0   

       Dribbling  Curve  FKAccuracy  LongPassing  BallControl  Acceleration  \
11537       65.0   65.0        67.0         59.0         75.0          59.0   
13132       58.0   51.0        31.0         30.0         64.0          52.0   
270         90.0   57.0        51.0         65.0         87.0          89.0   
13329       62.0   52.0        58.0         51.0         60.0          75.0   
1885        78.0   59.0        56.0         79.0         80.0          73.0   

       SprintSpeed  Agility  Reactions  Balance  ShotPower  Jumping  Stamina  \
11537         63.0     52.0       63.0     55.0       61.0     34.0     33.0   
13132         54.0     45.0       56.0     55.0       61.0     64.0     53.0   
270           92.0     88.0       77.0     82.0       78.0     57.0     76.0   
13329         77.0 

In [1120]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 14743 entries, 0 to 18206
Data columns (total 82 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Age                       14743 non-null  int64  
 1   Nationality               14743 non-null  object 
 2   Overall                   14743 non-null  int64  
 3   Potential                 14743 non-null  int64  
 4   Club                      14743 non-null  object 
 5   Value                     14743 non-null  object 
 6   Wage                      14743 non-null  object 
 7   Special                   14743 non-null  int64  
 8   Preferred Foot            14743 non-null  object 
 9   International Reputation  14743 non-null  float64
 10  Weak Foot                 14743 non-null  float64
 11  Skill Moves               14743 non-null  float64
 12  Work Rate                 14743 non-null  object 
 13  Body Type                 14743 non-null  object 
 14  Real F

Vemos que a pesar de que hay variables que se considerarían numéricas, se están reconociendo como 'object' por lo que vamos a enfocarnos en estas variables.

La primera de estas variables es 'Value'.

Esta representa el valor del mercado del jugador (un valor), pero dado a la forma en la que se guarda, se reconoce como un object, por lo que se va a transformar. Esto para facilidad de la construcción de cualquier modelo y el dashboard.

In [1121]:
df['Value'].value_counts().to_frame()

Value
€1.1M      360
€375K      305
€325K      290
€425K      284
€350K      273
...        ...
€44.5M       1
€56.5M       1
€53M         1
€55M         1
€110.5M      1

[210 rows x 1 columns]

Vemos que lo toma como un objeto dado que dentro del valor también se encuentra el signo euros y la cantidad, M para millones, K para miles.

Se procede a guardar esta variable únicamente con el valor numérico en Millones de euros.

In [1122]:
#Se aplica una función de transformación de los valores de la columna 'Value'
def label_value(row):
  value = str(row['Value'])
  length = len(value)
  num = value[1:length-1:1]
  #print(num)
  if value[0] == '€':
    if value[-1] == 'K':
      return float(num)/1000
    elif value[-1] == 'M':
      return num

df['Value'] = df.apply(lambda row: label_value(row), axis = 1)

In [1123]:
df['Value'].value_counts().to_frame()

Value
1.1      360
0.375    305
0.325    290
0.425    284
0.35     273
...      ...
44.5       1
56.5       1
53         1
55         1
110.5      1

[210 rows x 1 columns]

Vemos que los valores corresponden a antes de la transformación

In [1124]:
df['Value'] = df['Value'].astype(float)

In [1125]:
df.rename(columns = {'Value' : 'Value_in_Million'}, inplace = True)

La segunda columna que vamos a transformar es 'Wage'

Esta representa el salario de un jugador (un valor), pero dado a la forma en la que se guarda, se reconoce como un object, por lo que se va a transformar. Esto para facilidad de la construcción de cualquier modelo y el dashboard.

In [1126]:
df['Wage'].value_counts().to_frame()

Wage
€1K    3939
€2K    2406
€3K    1591
€4K    1049
€5K     690
...     ...
€300K     1
€405K     1
€250K     1
€255K     1
€565K     1

[141 rows x 1 columns]

In [1127]:
#Se aplica una función de transformación de los valores de la columna 'Wage'
def label_value(row):
  wage = str(row['Wage'])
  length = len(wage)
  num = wage[1:length-1:1]
  if(wage[0]) == '€':
    if wage[-1] == 'K':
      return num
    elif wage[-1] == 'M':
      return num*1000000

df['Wage'] = df.apply(lambda row: label_value(row), axis = 1)
df['Wage'].value_counts().to_frame()

Wage
1    3939
2    2406
3    1591
4    1049
5     690
..    ...
300     1
405     1
250     1
255     1
565     1

[141 rows x 1 columns]

In [1128]:
df['Wage'] = df['Wage'].astype(float)

In [1129]:
df.rename(columns = {'Wage' : 'Wage_in_Thousands'}, inplace = True)

El tercer atributo que representa un número pero el DF lo guarda como un 'object' es 'Contract Valid Until'

In [1130]:
df['Contract Valid Until'].value_counts().to_frame()

Contract Valid Until
2019                  4197
2021                  3873
2020                  3593
2022                  1340
2023                   947
2018                   766
2024                    20
2025                     5
2026                     2

Vemos que no hay anomalías, por lo que solo se convierte a int

In [1131]:
df['Contract Valid Until'] = df['Contract Valid Until'].astype(int)

El cuarto atributo que se guarda como object pero representa un número es 'Height'

In [1132]:
df['Height'].value_counts().to_frame()

Height
6'0     2375
5'10    2254
5'9     2058
5'11    1898
6'1     1462
6'2     1382
5'8      853
5'7      825
6'3      583
6'4      417
5'6      279
5'5      134
6'5      119
6'6       36
5'4       29
5'3       16
6'7       12
5'2        5
5'1        3
6'8        3

Se puede observar que no hay anomalias con los datos, por lo que se procede a transformar

In [1133]:
#Se aplica una función de transformación de los valores de la columna 'Height'
def label_value(row):
  height = row['Height'].split("'")
  #print(height)
  h = height[0] + "." + height[1]
  return float(h)
  

df['Height'] = df.apply(lambda row: label_value(row), axis = 1)
df['Height'].value_counts().to_frame()

Height
6.00    2375
5.10    2257
5.90    2058
5.11    1898
6.10    1462
6.20    1382
5.80     853
5.70     825
6.30     583
6.40     417
5.60     279
5.50     134
6.50     119
6.60      36
5.40      29
5.30      16
6.70      12
5.20       5
6.80       3

De esta manera, agrupamos los valores que estaban repetidos y no se tenían en cuenta, por ejemplo, el valor "5'1" y "5'10" que son el mismo, al cambiar su representación a un float, se toman ahora si como el mismo valor, esto no pasaba antes de hacer la transformación.

El quinto atributo que representa un número pero el DF lo guarda como un 'objec' es 'Weight'.

In [1134]:
df['Weight'].value_counts().to_frame()

Weight
165lbs    1273
154lbs    1272
159lbs     833
161lbs     814
172lbs     781
163lbs     777
176lbs     769
170lbs     718
168lbs     694
150lbs     615
174lbs     570
157lbs     552
143lbs     449
152lbs     448
181lbs     422
148lbs     393
179lbs     371
183lbs     360
146lbs     327
141lbs     299
187lbs     293
185lbs     286
139lbs     189
190lbs     177
194lbs     166
137lbs     157
192lbs     116
132lbs     110
196lbs      83
134lbs      78
198lbs      70
201lbs      38
203lbs      37
205lbs      29
128lbs      29
130lbs      27
207lbs      26
123lbs      18
209lbs      17
126lbs      14
212lbs       9
214lbs       8
121lbs       8
117lbs       6
119lbs       4
110lbs       2
216lbs       2
223lbs       2
218lbs       1
115lbs       1
243lbs       1
220lbs       1
234lbs       1

Se puede observar que esto es debido a que se guardan las unidades dentro. Gracias a la anterior tabla se puede ver que la unidad siempre es la misma, es decir, libras. Por lo que se procede a transformar este valor.

In [1135]:
#Se aplica una función de transformación de los valores de la columna 'Weight'
def label_value(row):
  weight = str(row['Weight'])
  w = weight[0:-3:1]
  #print(w)
  return int(w)
  
df['Weight'] = df.apply(lambda row: label_value(row), axis = 1)
df['Weight'].value_counts().to_frame()

Weight
165    1273
154    1272
159     833
161     814
172     781
163     777
176     769
170     718
168     694
150     615
174     570
157     552
143     449
152     448
181     422
148     393
179     371
183     360
146     327
141     299
187     293
185     286
139     189
190     177
194     166
137     157
192     116
132     110
196      83
134      78
198      70
201      38
203      37
205      29
128      29
130      27
207      26
123      18
209      17
126      14
212       9
214       8
121       8
117       6
119       4
110       2
216       2
223       2
218       1
115       1
243       1
220       1
234       1

In [1136]:
df.rename(columns = {'Weight' : 'Weight_in_lbs'}, inplace = True)

El conjunto de atributos que define la puntuación de un jugador por posición, es decir, el mismo cuya cantidad de nulos era 2085 al principio, guarda un valor numérico como una operación matemática, por lo que se procede a hacer una transformación.

In [1137]:
#Se aplica una función de transformación de los valores de la columna 'ST'
def label_value(row):
  st = row['ST'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['ST'] = df.apply(lambda row: label_value(row), axis = 1)
df['ST'].value_counts().to_frame()

ST
63  642
62  634
61  620
65  619
66  602
60  601
59  593
64  573
58  556
56  534
55  516
57  515
67  503
68  503
54  468
69  456
53  452
51  385
52  380
70  379
50  356
71  327
49  308
48  300
72  290
47  271
46  252
73  231
45  197
74  191
44  174
75  158
43  150
42  139
41  124
77  106
76   99
78   75
40   73
80   50
38   50
79   49
39   45
37   35
36   32
81   31
82   18
84   15
83   15
35   12
85    9
34    8
86    5
87    5
89    4
88    2
33    2
90    2
94    1
92    1

In [1138]:
#Se aplica una función de transformación de los valores de la columna 'LS'
def label_value(row):
  st = row['LS'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['LS'] = df.apply(lambda row: label_value(row), axis = 1)
df['LS'].value_counts().to_frame()

LS
63  642
62  634
61  620
65  619
66  602
60  601
59  593
64  573
58  556
56  534
55  516
57  515
67  503
68  503
54  468
69  456
53  452
51  385
52  380
70  379
50  356
71  327
49  308
48  300
72  290
47  271
46  252
73  231
45  197
74  191
44  174
75  158
43  150
42  139
41  124
77  106
76   99
78   75
40   73
80   50
38   50
79   49
39   45
37   35
36   32
81   31
82   18
84   15
83   15
35   12
85    9
34    8
86    5
87    5
89    4
88    2
33    2
90    2
94    1
92    1

In [1139]:
#Se aplica una función de transformación de los valores de la columna 'RS'
def label_value(row):
  st = row['RS'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['RS'] = df.apply(lambda row: label_value(row), axis = 1)
df['RS'].value_counts().to_frame()

RS
63  642
62  634
61  620
65  619
66  602
60  601
59  593
64  573
58  556
56  534
55  516
57  515
67  503
68  503
54  468
69  456
53  452
51  385
52  380
70  379
50  356
71  327
49  308
48  300
72  290
47  271
46  252
73  231
45  197
74  191
44  174
75  158
43  150
42  139
41  124
77  106
76   99
78   75
40   73
80   50
38   50
79   49
39   45
37   35
36   32
81   31
82   18
84   15
83   15
35   12
85    9
34    8
86    5
87    5
89    4
88    2
33    2
90    2
94    1
92    1

In [1140]:
#Se aplica una función de transformación de los valores de la columna 'LW'
def label_value(row):
  st = row['LW'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['LW'] = df.apply(lambda row: label_value(row), axis = 1)
df['LW'].value_counts().to_frame()

LW
65  673
63  640
61  628
64  623
62  622
67  616
66  604
68  569
59  563
60  556
57  530
69  497
58  494
70  480
56  479
71  399
72  367
54  365
55  365
53  327
52  314
51  300
73  300
74  249
50  248
49  219
75  213
76  202
47  193
48  182
45  152
44  142
46  137
43  137
77  135
42  126
41  114
40  111
79   91
39   87
78   87
38   83
37   74
36   59
80   55
81   46
35   45
82   38
83   37
34   34
84   26
33   26
85   23
32   13
87   10
86    9
31    6
90    5
89    5
88    4
92    3
29    2
94    1
30    1
91    1
27    1

In [1141]:
#Se aplica una función de transformación de los valores de la columna 'LF'
def label_value(row):
  st = row['LF'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['LF'] = df.apply(lambda row: label_value(row), axis = 1)
df['LF'].value_counts().to_frame()

LF
63  655
64  623
66  616
65  615
61  611
62  608
60  595
67  583
59  546
68  543
58  508
57  487
70  480
56  468
69  467
55  418
54  387
71  382
53  380
52  326
72  323
51  299
73  295
74  271
50  248
49  242
48  235
47  220
75  202
46  185
76  178
44  170
45  155
42  145
43  124
41  117
40  113
77  104
79   88
38   85
78   82
39   81
37   57
80   56
81   51
36   45
82   43
35   41
33   33
83   33
84   27
34   25
85   19
32   12
86   11
87    8
88    4
89    4
90    4
31    3
92    2
29    2
93    1
91    1
95    1

In [1142]:
#Se aplica una función de transformación de los valores de la columna 'CF'
def label_value(row):
  st = row['CF'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['CF'] = df.apply(lambda row: label_value(row), axis = 1)
df['CF'].value_counts().to_frame()

CF
63  655
64  623
66  616
65  615
61  611
62  608
60  595
67  583
59  546
68  543
58  508
57  487
70  480
56  468
69  467
55  418
54  387
71  382
53  380
52  326
72  323
51  299
73  295
74  271
50  248
49  242
48  235
47  220
75  202
46  185
76  178
44  170
45  155
42  145
43  124
41  117
40  113
77  104
79   88
38   85
78   82
39   81
37   57
80   56
81   51
36   45
82   43
35   41
33   33
83   33
84   27
34   25
85   19
32   12
86   11
87    8
88    4
89    4
90    4
31    3
92    2
29    2
93    1
91    1
95    1

In [1143]:
#Se aplica una función de transformación de los valores de la columna 'RF'
def label_value(row):
  st = row['RF'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['RF'] = df.apply(lambda row: label_value(row), axis = 1)
df['RF'].value_counts().to_frame()

RF
63  655
64  623
66  616
65  615
61  611
62  608
60  595
67  583
59  546
68  543
58  508
57  487
70  480
56  468
69  467
55  418
54  387
71  382
53  380
52  326
72  323
51  299
73  295
74  271
50  248
49  242
48  235
47  220
75  202
46  185
76  178
44  170
45  155
42  145
43  124
41  117
40  113
77  104
79   88
38   85
78   82
39   81
37   57
80   56
81   51
36   45
82   43
35   41
33   33
83   33
84   27
34   25
85   19
32   12
86   11
87    8
88    4
89    4
90    4
31    3
92    2
29    2
93    1
91    1
95    1

In [1144]:
#Se aplica una función de transformación de los valores de la columna 'RW'
def label_value(row):
  st = row['RW'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['RW'] = df.apply(lambda row: label_value(row), axis = 1)
df['RW'].value_counts().to_frame()

RW
65  673
63  640
61  628
64  623
62  622
67  616
66  604
68  569
59  563
60  556
57  530
69  497
58  494
70  480
56  479
71  399
72  367
54  365
55  365
53  327
52  314
51  300
73  300
74  249
50  248
49  219
75  213
76  202
47  193
48  182
45  152
44  142
46  137
43  137
77  135
42  126
41  114
40  111
79   91
39   87
78   87
38   83
37   74
36   59
80   55
81   46
35   45
82   38
83   37
34   34
84   26
33   26
85   23
32   13
87   10
86    9
31    6
90    5
89    5
88    4
92    3
29    2
94    1
30    1
91    1
27    1

In [1145]:
#Se aplica una función de transformación de los valores de la columna 'LAM'
def label_value(row):
  st = row['LAM'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['LAM'] = df.apply(lambda row: label_value(row), axis = 1)
df['LAM'].value_counts().to_frame()

LAM
63  683
64  650
60  626
61  624
62  621
65  617
66  590
67  586
59  559
68  551
58  520
69  498
57  497
56  462
70  438
55  410
71  392
54  384
53  364
72  359
52  316
73  290
51  282
50  256
74  244
49  234
48  231
75  228
47  191
76  176
44  155
45  154
46  154
77  139
42  132
43  114
40   97
41   92
78   92
39   80
79   65
81   62
37   61
38   58
82   50
80   49
36   48
83   39
34   34
84   32
35   31
33   27
32   26
85   15
86   15
31   10
87    8
89    7
90    5
88    4
30    3
92    2
91    2
95    1
29    1

In [1146]:
#Se aplica una función de transformación de los valores de la columna 'CAM'
def label_value(row):
  st = row['CAM'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['CAM'] = df.apply(lambda row: label_value(row), axis = 1)
df['CAM'].value_counts().to_frame()

CAM
63  683
64  650
60  626
61  624
62  621
65  617
66  590
67  586
59  559
68  551
58  520
69  498
57  497
56  462
70  438
55  410
71  392
54  384
53  364
72  359
52  316
73  290
51  282
50  256
74  244
49  234
48  231
75  228
47  191
76  176
44  155
45  154
46  154
77  139
42  132
43  114
40   97
41   92
78   92
39   80
79   65
81   62
37   61
38   58
82   50
80   49
36   48
83   39
34   34
84   32
35   31
33   27
32   26
85   15
86   15
31   10
87    8
89    7
90    5
88    4
30    3
92    2
91    2
95    1
29    1

In [1147]:
#Se aplica una función de transformación de los valores de la columna 'RAM'
def label_value(row):
  st = row['RAM'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['RAM'] = df.apply(lambda row: label_value(row), axis = 1)
df['RAM'].value_counts().to_frame()

RAM
63  683
64  650
60  626
61  624
62  621
65  617
66  590
67  586
59  559
68  551
58  520
69  498
57  497
56  462
70  438
55  410
71  392
54  384
53  364
72  359
52  316
73  290
51  282
50  256
74  244
49  234
48  231
75  228
47  191
76  176
44  155
45  154
46  154
77  139
42  132
43  114
40   97
41   92
78   92
39   80
79   65
81   62
37   61
38   58
82   50
80   49
36   48
83   39
34   34
84   32
35   31
33   27
32   26
85   15
86   15
31   10
87    8
89    7
90    5
88    4
30    3
92    2
91    2
95    1
29    1

In [1148]:
#Se aplica una función de transformación de los valores de la columna 'LM'
def label_value(row):
  st = row['LM'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['LM'] = df.apply(lambda row: label_value(row), axis = 1)
df['LM'].value_counts().to_frame()

LM
63  742
64  685
62  678
66  662
65  654
61  635
67  625
68  618
59  571
60  552
69  550
58  542
57  480
70  473
56  430
71  427
72  387
55  373
54  367
53  350
73  316
52  293
51  264
74  250
75  244
50  237
49  203
76  176
46  170
47  167
48  166
77  146
45  133
44  126
43  107
42  103
41   90
78   83
79   79
39   69
80   68
40   65
81   48
37   45
82   45
36   43
38   43
83   34
35   27
85   22
84   21
34   15
86   12
89    7
88    6
33    5
91    3
87    3
32    2
31    2
30    1
93    1
92    1
29    1

In [1149]:
#Se aplica una función de transformación de los valores de la columna 'LCM'
def label_value(row):
  st = row['LCM'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['LCM'] = df.apply(lambda row: label_value(row), axis = 1)
df['LCM'].value_counts().to_frame()

LCM
60  698
64  684
63  674
62  663
61  661
59  617
58  598
65  591
57  585
55  575
66  572
56  552
68  497
67  497
54  494
53  408
52  397
69  382
51  378
70  351
71  338
49  304
50  297
72  284
73  245
48  244
47  229
74  198
46  184
45  176
75  153
44  145
76  125
43  113
77  110
42   85
78   81
41   73
40   60
79   55
80   55
39   40
81   39
82   36
38   35
36   34
37   28
83   19
84   18
35   16
34   15
85   10
86    8
87    7
33    4
88    2
89    1
91    1
90    1
32    1

In [1150]:
#Se aplica una función de transformación de los valores de la columna 'CM'
def label_value(row):
  st = row['CM'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['CM'] = df.apply(lambda row: label_value(row), axis = 1)
df['CM'].value_counts().to_frame()

CM
60  698
64  684
63  674
62  663
61  661
59  617
58  598
65  591
57  585
55  575
66  572
56  552
68  497
67  497
54  494
53  408
52  397
69  382
51  378
70  351
71  338
49  304
50  297
72  284
73  245
48  244
47  229
74  198
46  184
45  176
75  153
44  145
76  125
43  113
77  110
42   85
78   81
41   73
40   60
79   55
80   55
39   40
81   39
82   36
38   35
36   34
37   28
83   19
84   18
35   16
34   15
85   10
86    8
87    7
33    4
88    2
89    1
91    1
90    1
32    1

In [1151]:
#Se aplica una función de transformación de los valores de la columna 'RCM'
def label_value(row):
  st = row['RCM'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['RCM'] = df.apply(lambda row: label_value(row), axis = 1)
df['RCM'].value_counts().to_frame()

RCM
60  698
64  684
63  674
62  663
61  661
59  617
58  598
65  591
57  585
55  575
66  572
56  552
68  497
67  497
54  494
53  408
52  397
69  382
51  378
70  351
71  338
49  304
50  297
72  284
73  245
48  244
47  229
74  198
46  184
45  176
75  153
44  145
76  125
43  113
77  110
42   85
78   81
41   73
40   60
79   55
80   55
39   40
81   39
82   36
38   35
36   34
37   28
83   19
84   18
35   16
34   15
85   10
86    8
87    7
33    4
88    2
89    1
91    1
90    1
32    1

In [1152]:
#Se aplica una función de transformación de los valores de la columna 'RM'
def label_value(row):
  st = row['RM'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['RM'] = df.apply(lambda row: label_value(row), axis = 1)
df['RM'].value_counts().to_frame()

RM
63  742
64  685
62  678
66  662
65  654
61  635
67  625
68  618
59  571
60  552
69  550
58  542
57  480
70  473
56  430
71  427
72  387
55  373
54  367
53  350
73  316
52  293
51  264
74  250
75  244
50  237
49  203
76  176
46  170
47  167
48  166
77  146
45  133
44  126
43  107
42  103
41   90
78   83
79   79
39   69
80   68
40   65
81   48
37   45
82   45
36   43
38   43
83   34
35   27
85   22
84   21
34   15
86   12
89    7
88    6
33    5
91    3
87    3
32    2
31    2
30    1
93    1
92    1
29    1

In [1153]:
#Se aplica una función de transformación de los valores de la columna 'LWB'
def label_value(row):
  st = row['LWB'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['LWB'] = df.apply(lambda row: label_value(row), axis = 1)
df['LWB'].value_counts().to_frame()

LWB
63  655
62  643
64  631
58  622
61  622
65  592
59  575
60  568
55  553
67  539
57  535
54  533
66  530
56  516
53  469
68  448
69  425
52  416
50  411
51  388
70  351
49  340
71  302
48  279
72  276
47  246
73  239
74  204
45  204
46  202
75  155
44  155
43  134
42  119
76  117
77  105
41  100
40   93
78   71
38   58
39   58
79   56
37   44
80   34
81   32
36   29
82   25
83   10
86    6
35    6
85    6
34    6
84    4
33    2
88    2
87    1
32    1

In [1154]:
#Se aplica una función de transformación de los valores de la columna 'LDM'
def label_value(row):
  st = row['LDM'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['LDM'] = df.apply(lambda row: label_value(row), axis = 1)
df['LDM'].value_counts().to_frame()

LDM
64  605
63  592
62  588
61  558
66  549
65  522
60  521
67  499
59  487
58  484
68  483
56  472
57  437
55  430
53  429
54  411
51  406
70  400
69  398
52  397
50  386
49  364
48  335
71  331
46  301
47  299
72  285
73  252
45  249
44  212
74  194
43  190
75  183
41  163
42  155
76  147
40  146
39  108
77   94
78   88
37   88
38   78
36   68
79   61
80   51
35   45
34   42
81   42
33   23
82   19
83   19
84   18
32   11
86    9
85    9
31    4
87    2
89    1
90    1
88    1
30    1

In [1155]:
#Se aplica una función de transformación de los valores de la columna 'CDM'
def label_value(row):
  st = row['CDM'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['CDM'] = df.apply(lambda row: label_value(row), axis = 1)
df['CDM'].value_counts().to_frame()

CDM
64  605
63  592
62  588
61  558
66  549
65  522
60  521
67  499
59  487
58  484
68  483
56  472
57  437
55  430
53  429
54  411
51  406
70  400
69  398
52  397
50  386
49  364
48  335
71  331
46  301
47  299
72  285
73  252
45  249
44  212
74  194
43  190
75  183
41  163
42  155
76  147
40  146
39  108
77   94
78   88
37   88
38   78
36   68
79   61
80   51
35   45
34   42
81   42
33   23
82   19
83   19
84   18
32   11
86    9
85    9
31    4
87    2
89    1
90    1
88    1
30    1

In [1156]:
#Se aplica una función de transformación de los valores de la columna 'RDM'
def label_value(row):
  st = row['RDM'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['RDM'] = df.apply(lambda row: label_value(row), axis = 1)
df['RDM'].value_counts().to_frame()

RDM
64  605
63  592
62  588
61  558
66  549
65  522
60  521
67  499
59  487
58  484
68  483
56  472
57  437
55  430
53  429
54  411
51  406
70  400
69  398
52  397
50  386
49  364
48  335
71  331
46  301
47  299
72  285
73  252
45  249
44  212
74  194
43  190
75  183
41  163
42  155
76  147
40  146
39  108
77   94
78   88
37   88
38   78
36   68
79   61
80   51
35   45
34   42
81   42
33   23
82   19
83   19
84   18
32   11
86    9
85    9
31    4
87    2
89    1
90    1
88    1
30    1

In [1157]:
#Se aplica una función de transformación de los valores de la columna 'RWB'
def label_value(row):
  st = row['RWB'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['RWB'] = df.apply(lambda row: label_value(row), axis = 1)
df['RWB'].value_counts().to_frame()

RWB
63  655
62  643
64  631
58  622
61  622
65  592
59  575
60  568
55  553
67  539
57  535
54  533
66  530
56  516
53  469
68  448
69  425
52  416
50  411
51  388
70  351
49  340
71  302
48  279
72  276
47  246
73  239
74  204
45  204
46  202
75  155
44  155
43  134
42  119
76  117
77  105
41  100
40   93
78   71
38   58
39   58
79   56
37   44
80   34
81   32
36   29
82   25
83   10
86    6
35    6
85    6
34    6
84    4
33    2
88    2
87    1
32    1

In [1158]:
#Se aplica una función de transformación de los valores de la columna 'LB'
def label_value(row):
  st = row['LB'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['LB'] = df.apply(lambda row: label_value(row), axis = 1)
df['LB'].value_counts().to_frame()

LB
63  616
65  613
61  612
60  606
62  592
66  589
64  553
67  526
59  523
58  505
56  503
57  478
68  464
55  463
53  454
69  442
54  439
51  404
52  403
50  367
48  347
70  346
49  337
47  299
72  289
46  277
71  275
45  236
73  230
44  216
43  203
74  187
42  162
75  136
41  135
40  133
76  106
77   97
39   97
38   86
78   73
37   65
79   49
36   49
80   35
35   31
82   24
81   22
34   15
83   10
85    7
87    5
33    5
84    3
86    2
32    1
31    1

In [1159]:
#Se aplica una función de transformación de los valores de la columna 'LCB'
def label_value(row):
  st = row['LCB'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['LCB'] = df.apply(lambda row: label_value(row), axis = 1)
df['LCB'].value_counts().to_frame()

LCB
64  582
65  579
66  547
67  537
63  525
68  493
62  461
61  448
70  447
69  432
60  402
59  391
48  349
58  347
47  339
53  336
46  330
71  323
57  320
72  318
56  317
49  315
55  315
54  309
43  307
45  307
44  303
50  302
52  292
42  284
73  276
51  272
74  255
39  245
40  238
41  237
38  199
75  186
37  168
76  154
36  137
35  123
77  115
34  100
78   78
79   65
33   63
32   51
81   39
80   30
83   30
82   27
31   22
84   18
85   13
30   13
86    9
87    8
29    8
88    3
90    2
89    1
27    1

In [1160]:
#Se aplica una función de transformación de los valores de la columna 'CB'
def label_value(row):
  st = row['CB'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['CB'] = df.apply(lambda row: label_value(row), axis = 1)
df['CB'].value_counts().to_frame()

CB
64  582
65  579
66  547
67  537
63  525
68  493
62  461
61  448
70  447
69  432
60  402
59  391
48  349
58  347
47  339
53  336
46  330
71  323
57  320
72  318
56  317
49  315
55  315
54  309
43  307
45  307
44  303
50  302
52  292
42  284
73  276
51  272
74  255
39  245
40  238
41  237
38  199
75  186
37  168
76  154
36  137
35  123
77  115
34  100
78   78
79   65
33   63
32   51
81   39
80   30
83   30
82   27
31   22
84   18
85   13
30   13
86    9
87    8
29    8
88    3
90    2
89    1
27    1

In [1161]:
#Se aplica una función de transformación de los valores de la columna 'RCB'
def label_value(row):
  st = row['RCB'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['RCB'] = df.apply(lambda row: label_value(row), axis = 1)
df['RCB'].value_counts().to_frame()

RCB
64  582
65  579
66  547
67  537
63  525
68  493
62  461
61  448
70  447
69  432
60  402
59  391
48  349
58  347
47  339
53  336
46  330
71  323
57  320
72  318
56  317
49  315
55  315
54  309
43  307
45  307
44  303
50  302
52  292
42  284
73  276
51  272
74  255
39  245
40  238
41  237
38  199
75  186
37  168
76  154
36  137
35  123
77  115
34  100
78   78
79   65
33   63
32   51
81   39
80   30
83   30
82   27
31   22
84   18
85   13
30   13
86    9
87    8
29    8
88    3
90    2
89    1
27    1

In [1162]:
#Se aplica una función de transformación de los valores de la columna 'RB'
def label_value(row):
  st = row['RB'].split('+')
  st2 = int(st[0]) + int(st[1])
  return int(st2)
  
df['RB'] = df.apply(lambda row: label_value(row), axis = 1)
df['RB'].value_counts().to_frame()

RB
63  616
65  613
61  612
60  606
62  592
66  589
64  553
67  526
59  523
58  505
56  503
57  478
68  464
55  463
53  454
69  442
54  439
51  404
52  403
50  367
48  347
70  346
49  337
47  299
72  289
46  277
71  275
45  236
73  230
44  216
43  203
74  187
42  162
75  136
41  135
40  133
76  106
77   97
39   97
38   86
78   73
37   65
79   49
36   49
80   35
35   31
82   24
81   22
34   15
83   10
85    7
87    5
33    5
84    3
86    2
32    1
31    1

Con esto, se han cambiado todos los atributos de la puntuación del jugador por posición de 'object' a 'int64' para poder hacer una mejor interpretación.

Por úlitmo, la última variable que guarda un valor numérico pero en el DF se guarda como 'object' es 'Release Clause'

In [1163]:
df['Release Clause'].value_counts().to_frame()

Release Clause
€1.1M              504
€1.3M              384
€1.2M              359
€1.4M              357
€1.5M              314
...                ...
€36.9M               1
€41.7M               1
€35.6M               1
€42.6M               1
€66K                 1

[1199 rows x 1 columns]

Se procede, al igual que con los atributos 'Wage' y 'Value', a transformarse para poder hacer una mejor interpretación.

In [1164]:
#Se aplica una función de transformación de los valores de la columna 'Release Clause'
def label_value(row):
  rc = str(row['Release Clause'])
  length = len(rc)
  num = rc[1:length-1:1]
  if rc[0] == '€':
    if rc[-1] == 'K':
      return float(num)/1000
    elif rc[-1] == 'M':
      return float(num)

df['Release Clause'] = df.apply(lambda row: label_value(row), axis = 1)
df['Release Clause'].value_counts().to_frame()

Release Clause
1.100              504
1.300              384
1.200              359
1.400              357
1.500              314
...                ...
36.900               1
41.700               1
35.600               1
42.600               1
0.066                1

[1199 rows x 1 columns]

In [1165]:
df.rename(columns = {'Release Clause' : 'Release Clause Million'}, inplace = True)

# **5. Visualización final**

Para terminar, se hace una última visualización del DF

In [1166]:
df.sample(5)

Age     Nationality  Overall  Potential                Club  \
11293   31  Korea Republic       64         64            FC Seoul   
9697    21         Nigeria       66         76            RC Celta   
13843   24    Saudi Arabia       62         66             Al Raed   
3016    25   United States       73         78  New York Red Bulls   
15276   22       Venezuela       60         69       CD Huachipato   

       Value_in_Million  Wage_in_Thousands  Special Preferred Foot  \
11293             0.300                2.0     1743          Right   
9697              1.100                5.0     1568          Right   
13843             0.400                4.0     1501          Right   
3016              4.500                6.0     1728          Right   
15276             0.325                1.0     1559          Right   

       International Reputation  Weak Foot  Skill Moves       Work Rate  \
11293                       1.0        3.0          2.0    High/ Medium   
9697                        1.0        3.0          2.0  Medium/ Medium   
13843                       1.0        2.0          3.0  Medium/ Medium   
3016                        1.0        3.0          2.0    Medium/ High   
15276                       1.0        3.0          2.0    High/ Medium   

      Body Type Real Face Position  Jersey Number        Joined  \
11293    Normal        No       RB           17.0   Jan 6, 2017   
9697     Normal        No       ST           35.0  Jul 11, 2018   
13843      Lean        No       ST           11.0  Aug 18, 2015   
3016     Normal       Yes      RCB           33.0  Feb 22, 2017   
15276    Normal        No       RM           11.0  Jan 27, 2018   

       Contract Valid Until  Height  Weight_in_lbs  LS  ST  RS  LW  LF  CF  \
11293                  2021    5.10            170  57  57  57  61  59  59   
9697                   2019    5.11            154  67  67  67  64  65  65   
13843                  2019    6.00            170  63  63  63  59  61  61   
3016                   2019    6.10            190  58  58  58  58  59  59   
15276                  2021    5.60            154  59  59  59  60  59  59   

       RF  RW  LAM  CAM  RAM  LM  LCM  CM  RCM  RM  LWB  LDM  CDM  RDM  RWB  \
11293  59  61   60   60   60  62   62  62   62  62   65   64   64   64   65   
9697   65  64   63   63   63  63   55  55   55  63   47   43   43   43   47   
13843  61  59   58   58   58  57   51  51   51  57   42   40   40   40   42   
3016   59  58   60   60   60  61   64  64   64  61   68   71   71   71   68   
15276  59  60   58   58   58  60   54  54   54  60   50   47   47   47   50   

       LB  LCB  CB  RCB  RB  Crossing  Finishing  HeadingAccuracy  \
11293  65   64  64   64  65      63.0       47.0             50.0   
9697   45   40  40   40  45      48.0       65.0             66.0   
13843  40   40  40   40  40      36.0       60.0             60.0   
3016   70   73  73   73  70      38.0       32.0             75.0   
15276  48   43  43   43  48      60.0       58.0             47.0   

       ShortPassing  Volleys  Dribbling  Curve  FKAccuracy  LongPassing  \
11293          64.0     58.0       58.0   51.0        49.0         61.0   
9697           63.0     65.0       67.0   54.0        36.0         38.0   
13843          49.0     46.0       61.0   52.0        37.0         34.0   
3016           71.0     36.0       58.0   36.0        34.0         66.0   
15276          57.0     55.0       59.0   57.0        42.0         58.0   

       BallControl  Acceleration  SprintSpeed  Agility  Reactions  Balance  \
11293         62.0          68.0         71.0     66.0       64.0     72.0   
9697          60.0          72.0         76.0     70.0       59.0     65.0   
13843         63.0          64.0         70.0     67.0       57.0     49.0   
3016          65.0          74.0         80.0     70.0       68.0     61.0   
15276         60.0          73.0         73.0     69.0       46.0     80.0   

       ShotPower  Jumping  Stami

In [1167]:
df.head()

Age Nationality  Overall  Potential                 Club  Value_in_Million  \
0   31   Argentina       94         94         FC Barcelona             110.5   
1   33    Portugal       94         94             Juventus              77.0   
2   26      Brazil       92         93  Paris Saint-Germain             118.5   
4   27     Belgium       91         92      Manchester City             102.0   
5   27     Belgium       91         91              Chelsea              93.0   

   Wage_in_Thousands  Special Preferred Foot  International Reputation  \
0              565.0     2202           Left                       5.0   
1              405.0     2228          Right                       5.0   
2              290.0     2143          Right                       5.0   
4              355.0     2281          Right                       4.0   
5              340.0     2142          Right                       4.0   

   Weak Foot  Skill Moves       Work Rate   Body Type Real Face Position  \
0        4.0          4.0  Medium/ Medium       Messi       Yes       RF   
1        4.0          5.0       High/ Low  C. Ronaldo       Yes       ST   
2        5.0          5.0    High/ Medium      Neymar       Yes       LW   
4        5.0          4.0      High/ High      Normal       Yes      RCM   
5        4.0          4.0    High/ Medium      Normal       Yes       LF   

   Jersey Number        Joined  Contract Valid Until  Height  Weight_in_lbs  \
0           10.0   Jul 1, 2004                  2021    5.70            159   
1            7.0  Jul 10, 2018                  2022    6.20            183   
2           10.0   Aug 3, 2017                  2022    5.90            150   
4            7.0  Aug 30, 2015                  2023    5.11            154   
5           10.0   Jul 1, 2012                  2020    5.80            163   

   LS  ST  RS  LW  LF  CF  RF  RW  LAM  CAM  RAM  LM  LCM  CM  RCM  RM  LWB  \
0  90  90  90  94  95  95  95  94   95   95   95  93   86  86   86  93   66   
1  94  94  94  92  93  93  93  92   91   91   91  91   84  84   84  91   68   
2  87  87  87  92  92  92  92  92   92   92   92  91   84  84   84  91   68   
4  85  85  85  90  90  90  90  90   91   91   91  91   90  90   90  91   80   
5  86  86  86  92  91  91  91  92   92   92   92  92   85  85   85  92   69   

   LDM  CDM  RDM  RWB  LB  LCB  CB  RCB  RB  Crossing  Finishing  \
0   63   63   63   66  61   49  49   49  61      84.0       95.0   
1   64   64   64   68  64   56  56   56  64      84.0       94.0   
2   63   63   63   68  63   50  50   50  63      79.0       87.0   
4   80   80   80   80  76   69  69   69  76      93.0       82.0   
5   66   66   66   69  63   52  52   52  63      81.0       84.0   

   HeadingAccuracy  ShortPassing  Volleys  Dribbling  Curve  FKAccuracy  \
0             70.0          90.0     86.0       97.0   93.0        94.0   
1             89.0          81.0     87.0       88.0   81.0        76.0   
2             62.0          84.0     84.0       96.0   88.0        87.0   
4             55.0          92.0     82.0       86.0   85.0        83.0   
5             61.0          89.0     80.0       95.0   83.0        79.0   

   LongPassing  BallControl  Acceleration  SprintSpeed  Agility  Reactions  \
0         87.0         96.0          91.0         86.0     91.0       95.0   
1         77.0         94.0          89.0         91.0     87.0       96.0   
2         78.0         95.0          94.0         90.0     96.0       94.0   
4         91.0         91.0          78.0         76.0     79.0       91.0   
5         83.0         94.0          94.0         88.0     95.0       90.0   

   Balance  ShotPower  Jumping  Stamina  Strength  LongShots  Aggression  \
0     95.0       85.0     68.0     72.0      59.0       94.0        48.0   
1     70.0       95.0     95.0     88.0      79.0       93.0        63.0   
2     84.0       80.0     61.0     81.0      49.0       82.0        56.0   
4     77.0       91.0     63.0 

In [1168]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 14743 entries, 0 to 18206
Data columns (total 82 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Age                       14743 non-null  int64  
 1   Nationality               14743 non-null  object 
 2   Overall                   14743 non-null  int64  
 3   Potential                 14743 non-null  int64  
 4   Club                      14743 non-null  object 
 5   Value_in_Million          14743 non-null  float64
 6   Wage_in_Thousands         14743 non-null  float64
 7   Special                   14743 non-null  int64  
 8   Preferred Foot            14743 non-null  object 
 9   International Reputation  14743 non-null  float64
 10  Weak Foot                 14743 non-null  float64
 11  Skill Moves               14743 non-null  float64
 12  Work Rate                 14743 non-null  object 
 13  Body Type                 14743 non-null  object 
 14  Real F

Con esto, se puede observar que los únicos atributos tipo 'object' son aquellos que no pueden tener una representación numérica. Y así finaliza el proceso de limpieza y depuración de los datos.

In [1169]:
df.to_csv("data_FIFA_limpiado.csv", sep=';', encoding='utf-8')